src/image.rs

```diff
@@ -511,6 +524,11 @@ pub struct Progress {
 }
 
 impl Progress {
+    /// Create Progress.
+    pub fn new(current: u64, total: u64) -> Self {
+        Self { current, total }
```

Comment:
Needed to add this function to actually create a progress object - Alternatively could make the fields public and use a struct constructor.

Reply:
Can you mark this `pub(crate)`? Currently there is one potential invariant (`current <= total`) that we may want to assert for instances of this type.

Reply:
Made it `pub(crate)`